src/Cupboard.Providers.MacOs/Bash/BashScriptProvider.cs

```diff
@@ -0,0 +1,144 @@
+using CliWrap;
+using CliWrap.EventStream;
+using Spectre.IO;
+
+namespace Cupboard
+{
+    public sealed class BashScriptProvider : AsyncMacResourceProvider<BashScript>
```

Comment:
Bash Script is not exclusive to Mac 🤔 

Windows has mingw bash and wsl bash.
Linux has bash :)

However, noted that you specified the CLI wrap for /bin/zsh which is default for mac.

Reply:
Correct.   It doesn't belong to windows either.  So I am open to suggestions on the taxonomy.  I put it here because I was adding macOS support and it's the default on mac as you stated.

Likely need some guidance from @patriksvensson on where he'd like it to live and be pulled into macOS.